cibuildwheel/linux.py

```diff
@@ -111,201 +102,133 @@ def build(options: BuildOptions) -> None:
+
+                if options.before_all:
+                    env = docker.get_environment()
+                    env['PATH'] = f'/opt/python/cp38-cp38:{env["PATH"]}'
+                    env = options.environment.as_dictionary(env, executor=docker.environment_executor)
+
+                    before_all_prepared = prepare_command(options.before_all, project='/project', package=container_package_dir)
+                    docker.call(['sh', '-c', before_all_prepared], env=env)
+
+                for config in platform_configs:
+                    dependency_constraint_flags: List[Union[str, PathLike]] = []
+
+                    if options.dependency_constraints:
+                        constraints_file = options.dependency_constraints.get_for_python_version(config.version)
+                        container_constraints_file = PurePath('/constraints.txt')
+
+                        docker.copy_into(constraints_file, container_constraints_file)
+                        dependency_constraint_flags = ['-c', container_constraints_file]
+
+                    env = docker.get_environment()
+
+                    # put this config's python top of the list
+                    python_bin = config.path / 'bin'
+                    env['PATH'] = f'{str(python_bin)}:{env["PATH"]}'
+
+                    env = options.environment.as_dictionary(env, executor=docker.environment_executor)
+
+                    # check config python and pip are still on PATH
+                    which_python = docker.call(['which', 'python'], env=env, capture_output=True).strip()
+                    if PurePath(which_python) != python_bin / 'python':
+                        print("cibuildwheel: python available on PATH doesn't match our installed instance. If you have modified PATH, ensure that you don't overwrite cibuildwheel's entry or insert python above it.", file=sys.stderr)
+                        exit(1)
+
+                    which_pip = docker.call(['which', 'pip'], env=env, capture_output=True).strip()
+                    if PurePath(which_pip) != python_bin / 'pip':
+                        print("cibuildwheel: pip available on PATH doesn't match our installed instance. If you have modified PATH, ensure that you don't overwrite cibuildwheel's entry or insert pip above it.", file=sys.stderr)
+                        exit(1)
+
+                    if options.before_build:
+                        before_build_prepared = prepare_command(options.before_build, project='/project', package=container_package_dir)
+                        docker.call(['sh', '-c', before_build_prepared], env=env)
+
+                    temp_dir = PurePath('/tmp/cibuildwheel')
+                    built_wheel_dir = temp_dir / 'built_wheel'
+                    docker.call(['rm', '-rf', built_wheel_dir])
+                    docker.call(['mkdir', '-p', built_wheel_dir])
+
+                    docker.call(
+                        [
```

Comment:
Same here, as in the other PR. Feels like very broad formatting, but maybe that's fine.

Reply:
👍